oxide-auth-actix/examples/actix-example/src/main.rs

```diff
@@ -0,0 +1,236 @@
+                }),
+            )
+            .route(
+                "/",
+                web::get().to_async(|(req, state): (OAuthResource, web::Data<Addr<State>>)| {
+                    state
+                        .send(Resource(req.into_request()).wrap(None))
+                        .map_err(WebError::from)
+                        .and_then(|res| match res {
+                            Ok(_grant) => Ok(OAuthResponse::ok()
+                                .content_type("text/plain")?
+                                .body("Hello world!")),
+                            Err(Ok(response)) => Ok(response.body(DENY_TEXT)),
+                            Err(Err(e)) => Err(e.into()),
+                        })
+                }),
+            )
+    })
+    .bind("localhost:8020")
+    .expect("Failed to bind to socket")
+    .start();
+
+    support::dummy_client();
+
+    // Start, then open in browser, don't care about this finishing.
+    let _: Result<(), ()> = sys.block_on(future::lazy(|| {
+        let _ = thread::spawn(support::open_in_browser);
+        future::ok(())
+    }));
+
+    // Run the rest of the system.
+    let _ = sys.run();
+}
+
+impl State {
+    pub fn preconfigured() -> Self {
+        State {
+            // A registrar with one pre-registered client
+            registrar: vec![Client::public(
+                "LocalClient",
+                "http://localhost:8021/endpoint".parse().unwrap(),
+                "default-scope".parse().unwrap(),
+            )]
+            .into_iter()
+            .collect(),
+            // Authorization tokens are 16 byte random keys to a memory hash map.
+            authorizer: AuthMap::new(RandomGenerator::new(16)),
+            // Bearer tokens are also random generated but 256-bit tokens, since they live longer
+            // and this example is somewhat paranoid.
+            //
+            // We could also use a `TokenSigner::ephemeral` here to create signed tokens which can
+            // be read and parsed by anyone, but not maliciously created. However, they can not be
+            // revoked and thus don't offer even longer lived refresh tokens.
+            issuer: TokenMap::new(RandomGenerator::new(16)),
+
+            // A single scope that will guard resources for this endpoint
+            scopes: vec!["default-scope".parse().unwrap()],
+        }
+    }
+
+    pub fn with_solicitor<'a, S>(
+        &'a mut self,
+        solicitor: S,
+    ) -> impl Endpoint<OAuthRequest, Error = WebError> + 'a
+    where
+        S: OwnerSolicitor<OAuthRequest> + 'static,
+    {
+        EndpointErrorMapper::new(Generic {
+            authorizer: &mut self.authorizer,
+            registrar: &mut self.registrar,
+            issuer: &mut self.issuer,
+            solicitor,
+            scopes: self.scopes.clone(),
+            response: OAuthResponse::ok,
+        })
+    }
+}
+
+impl Endpoint<OAuthRequest> for State {
```

Comment:
If the other code also uses `Generic`, *maybe* switch to it for other code flow as well.

What are the pros and cons of having an implementation of `Endpoint` in the example?

Reply:
I like not needing to name all the types on Generic all the time. I supposed I could just `Box<dyn Endpoiint<OAuthRequest, Error = WebError>>`, though

Reply:
You can utilize [`Vacant`](https://docs.rs/oxide-auth/0.4.5/oxide_auth/frontends/simple/endpoint/struct.Vacant.html) to return a `Generic` without a solicitor but still annotate the return type as 
> `-> impl Endpoint<OAuthRequest, Error=WebError>` 

```
Generic {
…
    solicitor: Vacant,
}
```

Reply:
Well, technically it then has a solicitor that denies everything but that should not make a functional difference where it is used.